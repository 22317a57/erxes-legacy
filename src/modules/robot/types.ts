export type IFeature = {
  name: string;
  text: string;
};
