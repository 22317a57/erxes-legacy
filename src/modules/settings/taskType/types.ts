export interface ITaskType {
  _id: string;
  name: string;
  icon: string;
}

// query types

export type TaskTypeQueryResponse = {
  taskTypes: ITaskType[];
  loading: boolean;
  refetch: () => void;
};

export type MutationVariables = {
  _id?: string;
  name?: string;
  icon?: string;
};

// mutation types

export type AddMutationResponse = {
  addMutation: (mutation: { variables: MutationVariables }) => Promise<any>;
};

export type EditMutationResponse = {
  editMutation: (mutation: { variables: MutationVariables }) => Promise<any>;
};

export type RemoveMutationResponse = {
  removeMutation: (mutation: { variables: { _id: string } }) => Promise<any>;
};
