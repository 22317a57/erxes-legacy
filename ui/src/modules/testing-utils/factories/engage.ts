import * as Factory from 'factory.ts';
import { IAttachment } from 'modules/common/types';
import { IEmailDelivery, IEngageEmail, IEngageMessage, IEngageMessenger, IEngageScheduleDate, IEngageStats } from 'modules/engage/types';
import { ISegment, ISegmentCondition } from 'modules/segments/types';
import { IEmailSignature } from 'modules/settings/email/types';
import { brandFactory } from './brands';
import { conditionsRuleFactory } from './common';
import { tagFactory } from './tags';
import { userFactory } from './user';

export const engageMessengerFactory = Factory.Sync.makeFactory<IEngageMessenger>({
  brandId: '22',
  kind: 'auto',
  sentAs: 'Dorj',
  content: 'Hello',
  rules: [
    conditionsRuleFactory.build(),
    conditionsRuleFactory.build({ _id: '21' })
  ]
});

export const attachmentFactory = Factory.Sync.makeFactory<IAttachment>({
  name: 'Images1',
  type: 'img',
  url: '/images/logo',
  size: 12
});

export const engageEmailFactory = Factory.Sync.makeFactory<IEngageEmail>({
  templateId: '3',
  subject: 'Subject',
  content: 'content',
  attachments: [
    attachmentFactory.build(),
    attachmentFactory.build({ name: 'Images1' })
  ],
});

export const segmentConditionFactory = Factory.Sync.makeFactory<ISegmentCondition>({
  _id: '15',
  field: 'auto',
  value: 'string',
  operator: 'string',
  dateUnit: 'string',
  type: 'string',
  brandId: 'string',
});

export const segmentFactory = Factory.Sync.makeFactory<ISegment>({
  _id: '5',
  contentType: 'type',
  getSubSegments: [],
  getParentSegment: {} as ISegment,
  // ISegmentDocs
  name: 'segment',
  description: 'desc',
  color: '#fff',
  connector: 'connect',
  conditions: [
    segmentConditionFactory.build(),
    segmentConditionFactory.build({ _id: '15' })
  ],
  subOf: 'sub'
});

export const emailSignatureFactory = Factory.Sync.makeFactory<IEmailSignature>({
  brandId: '22',
  signature: 'erxes'
});

export const engageStatsFactory = Factory.Sync.makeFactory<IEngageStats>({
  send: 3,
  delivery: 4,
  open: 5,
  click: 11,
  complaint: 1,
  bounce: 6,
  renderingfailure: 7,
  reject: 9,
  total: 25,
});

export const engageScheduleDateFactory = Factory.Sync.makeFactory<IEngageScheduleDate>({
  type: 'everyday',
  month: 'July',
  day: '23',
  time: new Date()
});

export const engageMessageFactory = Factory.Sync.makeFactory<IEngageMessage>({
  _id: '4',
  stopDate: new Date(),
  createdDate: new Date(),
  messengerReceivedCustomerIds: ['test', '4'],
  brand: brandFactory.build({ _id: '3', name: 'Erxes' }),
  segment: segmentFactory.build({ _id: '5' }),
  fromUser: userFactory.build({ _id: '12', email: 'erxes@nmma.co' }),
  tagIds: ['12', '13'],
  getTags: [
    tagFactory.build(),
    tagFactory.build({ _id: '23' })
  ],
  stats: engageStatsFactory.build({}),
  logs: [{ message: 'string' }],
  // IEngageMessageDocs
  kind: 'auto',
  type: 'email',
  segmentIds: ['1,2'],
  brandIds: ['3,4'],
  customerIds: ['3,4'],
  title: 'test',
  fromUserId: '22',
  method: '22',
  isDraft: true,
  isLive: false,
  email: engageEmailFactory.build({ templateId: '3' }),
  messenger: engageMessengerFactory.build({ brandId: '22' }),
  scheduleDate: engageScheduleDateFactory.build({ type: 'everyday', month: 'July', day: '12', time: '2' })
});

export const emailDeliveryFactory = Factory.Sync.makeFactory<IEmailDelivery>({
  _id: '33',
  subject: 'subject',
  body: 'Body',
  to: 'Dulam@yahoo.com',
  cc: 'erxes@gmail.com',
  bcc: 'erxes@nmma.co',
  attachments: [JSON],
  from: 'erxes',
  kind: 'auto',
  userId: '12',
  customerId: '22',

  fromUser: userFactory.build({ _id: '12' }),
  fromEmail: 'erxes@nmma.co'
});