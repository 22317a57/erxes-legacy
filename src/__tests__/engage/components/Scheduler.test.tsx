// test passed
import { mount, shallow } from 'enzyme';
import Scheduler from 'modules/engage/components/Scheduler';
import { IEngageScheduleDate } from 'modules/engage/types';
import * as React from 'react';
import * as renderer from 'react-test-renderer';

describe('Scheduler component', () => {
  const testIEngageScheduleDate: IEngageScheduleDate = {
    type: 'string',
    month: 'string',
    day: 'string',
    time: new Date()
  };

  const defaultProps = {
    scheduleDate: testIEngageScheduleDate,
    onChange: (name: 'scheduleDate', value: IEngageScheduleDate) => null
  };

  test('renders successfully', () => {
    shallow(<Scheduler {...defaultProps} />);
  });

  test('renders with default props', () => {
    const control = mount(<Scheduler {...defaultProps} />);
    const props = control.props();

    expect(props).toMatchObject(defaultProps);
  });

  test('snapshot matches', () => {
    const rendered = renderer.create(<Scheduler {...defaultProps} />).toJSON();

    expect(rendered).toMatchSnapshot();
  });
});
