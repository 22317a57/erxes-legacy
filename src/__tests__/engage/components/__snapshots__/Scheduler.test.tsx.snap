// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Scheduler component snapshot matches 1`] = `
<div
  className="sc-jKVCRD eDSGVr"
>
  <label
    className="sc-ebFjAB bHuvQk"
  >
    Schedule:
  </label>
  <div
    className="sc-LKuAh jQIRMK"
  >
    <select
      className="sc-hzNEM bWROUb sc-iBEsjs jXYbvT"
      disabled={false}
      onChange={[Function]}
      required={false}
      value="string"
    >
      <option />
       
      <option
        value="day"
      >
        Every Day
      </option>
      <option
        value="month"
      >
        Every Month
      </option>
      <option
        value="year"
      >
        Every Year
      </option>
      <option
        value={1}
      >
        Every Monday
      </option>
      <option
        value={2}
      >
        Every Tuesday
      </option>
      <option
        value={3}
      >
        Every Wednesday
      </option>
      <option
        value={4}
      >
        Every Thursday
      </option>
      <option
        value={5}
      >
        Every Friday
      </option>
      <option
        value={6}
      >
        Every Saturday
      </option>
      <option
        value={0}
      >
        Every Sunday
      </option>
    </select>
  </div>
  <div
    className="sc-bSbAYC gPitej"
  />
  <div
    className="sc-guztPN bXvTbB"
  >
    <label
      className="sc-ebFjAB bHuvQk"
    >
      Choose time:
    </label>
    <div
      className="rdt rdtStatic rdtOpen"
    >
      <div
        className="rdtPicker"
      >
        <div
          className="rdtTime"
        >
          <table>
            <tbody>
              <tr>
                <td>
                  <div
                    className="rdtCounters"
                  >
                    <div
                      className="rdtCounter"
                    >
                      <span
                        className="rdtBtn"
                        onContextMenu={[Function]}
                        onMouseDown={[Function]}
                      >
                        ▲
                      </span>
                      <div
                        className="rdtCount"
                      >
                        21
                      </div>
                      <span
                        className="rdtBtn"
                        onContextMenu={[Function]}
                        onMouseDown={[Function]}
                      >
                        ▼
                      </span>
                    </div>
                    <div
                      className="rdtCounterSeparator"
                    >
                      :
                    </div>
                    <div
                      className="rdtCounter"
                    >
                      <span
                        className="rdtBtn"
                        onContextMenu={[Function]}
                        onMouseDown={[Function]}
                      >
                        ▲
                      </span>
                      <div
                        className="rdtCount"
                      >
                        55
                      </div>
                      <span
                        className="rdtBtn"
                        onContextMenu={[Function]}
                        onMouseDown={[Function]}
                      >
                        ▼
                      </span>
                    </div>
                  </div>
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>
  </div>
</div>
`;
